pkg/k8shandler/fluentd.go

```diff
@@ -416,7 +416,7 @@ func newFluentdPodSpec(cluster *logging.ClusterLogging, elasticsearchAppName str
 	// Shorten the termination grace period from the default 30 sec to 10 sec.
 	fluentdPodSpec.TerminationGracePeriodSeconds = utils.GetInt64(10)
 
-	if !pipelineSpec.DisableDefaultForwarding {
+	if !pipelineSpec.DisableDefaultForwarding && cluster.Spec.LogStore != nil {
```

Comment:
what happens when someone who already logging deployed and thus already got the logstore deployed, and then upgrades to this version?  Do their logstore resources get abandoned in place?  Do they need to manually clean them  up?

Reply:
If someone deploys cluster-logging with the default logstore with/without using LogForwarding and upgrades to this version does not abandon the installed log store. Entering this code branch means that the user intentionally removes all log store references from the ClusterLogging resource before/after upgrading.